### PR TITLE
Add arm64 simulator slice to binary builds.

### PR DESCRIPTION
**Pull request by ryanwilson**
This re-architects things to split by platform and build for
architectures together instead of having to lipo them together. Still
need to fix Carthage and test a full release.

### CODE REVIEW

File: ReleaseTooling/Sources/ZipBuilder/CocoaPodUtils.swift
Comment (ryanwilson) [inline ReleaseTooling/Sources/ZipBuilder/CocoaPodUtils.swift:547]: I believe this is what is causing the occasional failure on CI - I ran into it locally once by chance.

Comment (paulb777) [reply ReleaseTooling/Sources/ZipBuilder/CocoaPodUtils.swift:547]: Hmm, I wonder how that could ever happen?

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:306]: Needs testing that dynamic frameworks still get built right?

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:593]: s/file/folder/ ?

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:701]: s/exist/exists/

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:661]: This is something that really could use a unit test ... but not right now

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:874]: TODO

**Review by paulb777 — COMMENTED** [review]
Nice improvements!

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (ryanwilson) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:661]: Agreed 😄

**Comment by ryanwilson** [comment]
Fixing CI, please wait for review.

**Comment by ryanwilson** [comment]
Fixed.

**Review by paulb777 — APPROVED** [review]
Thanks!

**Comment by paulb777** [comment]
Ideally, we should understand the Database and Firestore cron failures before making a release with this

**Comment by ryanwilson** [comment]
> Ideally, we should understand the Database and Firestore cron failures before making a release with this

SGTM, I'll get working on those now and reach out to Gran.

**Comment by paulb777** [comment]
Should be good to go after a clean CI run of #6986

**Comment by ryanwilson** [comment]
I'm going to land this to unblock building a new RC to include arm64 slices for the simulator, and will update it if I see any issues the the generated artifact.

**Pull request merged**
